parser/prism/Parser.cc

```diff
@@ -7,12 +7,12 @@ using namespace std;
 
 namespace sorbet::parser::Prism {
 
-unique_ptr<parser::Node> Parser::run(core::GlobalState &gs, core::FileRef file, bool directlyDesugar) {
-    auto source = file.data(gs).source();
+unique_ptr<parser::Node> Parser::run(core::MutableContext &ctx, core::FileRef file, bool directlyDesugar) {
```

Comment:
Could you remove the `file` parameter here? It's redundant with the use of `MutableContext` now.

Reply:
This is pretty convenient. Now the Parser doesn't parse a file, but a context, which reads as a little unusually.

On the fence about it. It's nice to simplify the Translator constructors and fields though. LMK what you think

Reply:
You could construct the `MutableContext` inside of `Parser::run` instead. That's a little more idiomatic WRT the rest of the codebase. I'd advise against passing both a `MutableContext` and a file, when the two files are expected to be the same; passing both a file and a `MutableContext` just opens the door to accidental mismatches.

Reply:
Added note in https://github.com/sorbet/sorbet/issues/9065